platform/core.multiview/src/org/netbeans/core/multiview/MultiViewPeer.java

```diff
@@ -801,11 +799,11 @@ private UndoRedo privateGetUndoRedo() {
      * Delegates to CloseOperationHandler.
      */
     boolean canClose() {
-        Collection col = model.getCreatedElements();
-        Iterator it = col.iterator();
-        Collection badOnes = new ArrayList();
+        Collection<MultiViewElement> col = model.getCreatedElements();
+        Iterator<MultiViewElement> it = col.iterator();
+        Collection<CloseOperationState> badOnes = new ArrayList<>();
         while (it.hasNext()) {
```

Comment:
Should this be a nested for loop not to while loops?

Reply:
It really shouldn't. They could be converted to enhanced-for-loops but they shouldn't be nested.
The first while loop collects all non-closeable MVEs which the second while loop - if the closeHandler indicated such - attemps to close again.

You know the dialog showing you the unsaved files when you attempt to close one editor or the whole IDE. That's the principle here.